American Whitewater/ViewControllers/Map/MapViewController.swift

```diff
@@ -146,18 +121,11 @@ class MapViewController: UIViewController, MKMapViewDelegate, CLLocationManagerD
         // add the reaches from core data to the map
         if let reaches = fetchedResultsController?.fetchedObjects {
             mapView.removeAnnotations(mapView.annotations)
-
             mapView.addAnnotations(reaches)
             
-            // update the zoom if we're not using location
-            if DefaultsManager.showDistanceFilter == false {
-                let cleanedAnnotations = mapView.annotations.filter { $0.coordinate.latitude > 0 && $0.coordinate.longitude > -170 }
-                if MapViewController.userChangedMap == false {
-                    mapView.fitAll(in: cleanedAnnotations, andShow: true)
-                } else {
-                    mapView.addAnnotations(cleanedAnnotations)
-                }
-            }
+            // Zoom to updated coordinates
+            let cleanedAnnotations = mapView.annotations.filter { $0.coordinate.latitude > 0 && $0.coordinate.longitude > -170 }
+            mapView.fitAll(in: cleanedAnnotations, andShow: true)
```

Comment:
Can fitAll(in:andShow:) ever get called with an empty array? (It will call setVisibleMapRect() with an invalid rect if so)

Reply:
That's a great point, I'm sure there's a reach somewhere with no points.